File: 21560 - Sai Suraj/week 1/1614_Maximum Nesting Depth of the Parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        curr_depth = depth = 0
        for c in s:
            if c=="(":
                curr_depth +=1
                
            if c==")":
                depth = max(depth,curr_depth)
                curr_depth-=1
        return depth

File: 21558 - Nithin/week 1/1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int depth = 0, count = 0;
        int length = s.length();
        
        if(length == 0 || (length == 1 && (s[0] != '(' || s[0] != ')')))
            return depth;
        
        stack<char> stck;
        for(int i = 0; i < length; i++){
            if(s[i] == '('){
                stck.push(s[i]);
                ++count;
                depth = max(count, depth);
            }
            else if(s[i] == ')' && stck.top() == '('){
                stck.pop();
                --count;
                depth = max(count, depth);
            }
            else
                continue;
        }
        return depth;
    }
};